Add blogpost about sprockets to webpack migration

arielj opened the pull request:
**IMPORTANT: blog posts must be approved by two reviewers before they can be published, one of which must be a native English speaker.**

This is the PR for the blog post based on the work I did to migrate the Styleguide and FastRuby.io to support Webpack.

arielj commented:
@lubc I wasn't sure which categories to use (tech-debt? rails? webpack?) maybe some categories work better?

rdormer commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,309 @@
+
+### Reorganization and Reconfiguration
+
+The Webpacker gem expects files in different locations than Sprockets, and the configuration is different with different concepts an options.
```
an => and

rdormer commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
````diff
@@ -0,0 +1,309 @@
+@import "vendor/assets/stylesheets/fastruby/styleguide-core.scss";
+```
+
+This is really similar, just replaced the new package's names and locations for each file we need.
````
just replaced => "we just replaced" or "just replacing"

rdormer commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,309 @@
+
+## Conclusion
+
+It takes some time and a bit of a mindset change since Sprockets and Webpack works differently, but now are using the current Rails standard to handle assets and we can use modern JS features, frameworks, and tools more easily.
```
I would consider splitting this sentence up.  As is, it's a run-on sentence.

rdormer reviewed (changes requested):
a few small edits, but overall this looks good

arielj commented:
@rdormer thanks! just fixed your feedback

etagwerker commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,311 @@
+
+### FastRuby.io Styleguide
+
+We use our styleguide for the base style of our projects, but it works either as a Ruby gem or as static files. In order to use it with Webpack, we need to add support to be used as a node module (we'll use Yarn since it's the default node modules manager for Rails applications -like bundler for gems management-).
```
@arielj Maybe add a link to yarn?

etagwerker commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,311 @@
+
+### 3rd Party Assets in Gems
+
+Some assets are added by gems, but, similar to the styleguide, it's easier to integrate them with Webpack if we use an equivalent Yarn package.
```
@arielj Could you add an example? It might make it clearer to understand. 

Are you talking about the difference in including 3rd party libraries? For instance: If you use sprockets, then you would use something like the bootstrap-rails gem, but if you are using webpacker then you need to find the right npm package, is that it?

arielj replied:
Yes, that's what I meant, I'll add `@rails/ujs` as an example since I used that package during the actual steps.

etagwerker commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,311 @@
+
+## Linking the JavaScript
+
+`Sprocket-rails` gem provides the `javascript_include_tag` helper to add the `<script>` tag in the head. `Webpacker` gem provides a similar helper: `javascript_pack_tag`.
```
```suggestion
`sprocket-rails` gem provides the `javascript_include_tag` helper to add the `<script>` tag in the head. `Webpacker` gem provides a similar helper: `javascript_pack_tag`.
```

etagwerker commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,311 @@
+
+`Sprocket-rails` gem provides the `javascript_include_tag` helper to add the `<script>` tag in the head. `Webpacker` gem provides a similar helper: `javascript_pack_tag`.
+
+We'll use both files simultaneously while migrating and remove the previous one at the end, we we do incremental changes.
```
```suggestion
We'll use both files simultaneously while migrating. At the end we will remove the previous one, that way we test incremental changes.
```

etagwerker commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
````diff
@@ -0,0 +1,311 @@
+...
+```
+
+### Third-party Code
````
In the interest of being consistent about how we write 3rd:

```suggestion
### 3rd Party Code
```

etagwerker commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
````diff
@@ -0,0 +1,311 @@
+require('@rails/ujs').start()
+```
+
+Each third-party package that you use can have its own way to use it with Webpack.
````
```suggestion
Each 3rd party package that you use can have its own way to use it with Webpack.
```

etagwerker reviewed (commented):
@arielj Great article! Just added a few comments.

Also, maybe you can include a link to this PR: https://github.com/fastruby/styleguide/pull/23/files -- in case people want to dig deeper.

arielj commented:
> @arielj Great article! Just added a few comments.

thanks! just sent a PR addressing those comments

> 
> Also, maybe you can include a link to this PR: https://github.com/fastruby/styleguide/pull/23/files -- in case people want to dig deeper.

That PR is only the styleguide's changes, most of the migration changes are for FastrRuby.io but it's private. It might be confusing to link only to that one (maybe if in the future we make FastRuby.io an public project?).

etagwerker commented:
@arielj In the article you describe both changes that's why I thought it was relevant. When you link it make sure to say that those are the changes that were necessary for the style guide. 😉

It can be in a section at the very last with Resources to make it less confusing.

etagwerker reviewed (approved):
Looks good 👍

lubc commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,315 @@
+
+### CSS Entry Point
+
+Sprockets can find assets inside the node modules folder, so we are going to provide the SASS entry point.
```
```suggestion
Sprockets can find assets inside the node_modules folder, so we are going to provide the SASS entry point.
```

lubc commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,315 @@
+
+### Init a Yarn Package
+
+Inside the Styleguide's project folder, we run the command `yarn init`. This starts a wizard to setup the metadata of the project (name, git repository, author, license, and many more). At the end, creates a `package.json` file where we can configure this projects.
```
```suggestion
Inside the Styleguide's project folder, we run the command `yarn init`. This starts a wizard to set up the metadata of the project (name, git repository, author, license, and many more). At the end, creates a `package.json` file where we can configure this project.
```

lubc commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,315 @@
+
+### 3rd Party Code
+
+We use rails-ujs to handle the remote form submission and other Rails event. Instead of moving files, for this case we have to add the `@rails/ujs` package and initialize it inside our `application.js`.
```
```suggestion
We use rails-ujs to handle the remote form submission and other Rails events. Instead of moving files, for this case we have to add the `@rails/ujs` package and initialize it inside our `application.js`.
```

lubc commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,315 @@
+
+## Configure CircleCI
+
+Since we are using CircleCI as a continuos integration service an we were not using Yarn to handle 3rd party node modules before, we need to fix the configuration so CircleCI installs all Yarn packages before running the tests.
```
```suggestion
Since we are using CircleCI as a continuous integration service and we were not using Yarn to handle 3rd party node modules before, we need to fix the configuration so CircleCI installs all Yarn packages before running the tests.
```

lubc commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,315 @@
+
+## Conclusion
+
+It takes some time and a bit of a mindset change since Sprockets and Webpack works differently, but now we are using the current Rails standard to handle assets. Thanks to this and Webpack's popularity, we can now use modern JS features, frameworks, and tools more easily.
```
```suggestion
It takes some time and a bit of a mindset change since Sprockets and Webpack work differently, but now we are using the current Rails standard to handle assets. Thanks to this and Webpack's popularity, we can now use modern JS features, frameworks, and tools more easily.
```

lubc commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,315 @@
+date: 2020-10-09 11:00:00
+reviewed: 2020-10-09 11:00:00
+categories: ["rails", "webpack"]
+author: "arielj"
```
You were added as `arieljuod` instead

```suggestion
author: "arieljuod"
```

lubc commented on _posts/2020-10-09-from-sprockets-to-webpacker.markdown:
```diff
@@ -0,0 +1,315 @@
+---
+layout: post
+title: "From Sprockets to Webpacker"
```
Just a suggestion:

```suggestion
title: "How to migrate from Sprockets to Webpacker"
```

arielj replied:
Maybe that sounds like we are migrating all assets from sprockets to webpacker (even CSS, images, etc)? We are only moving the JS.

arielj replied:
Maybe `How to Migrate your JavaScript from Sprockets to Webpacker`? maybe it's too long

lubc replied:
@arielj sounds good 👍 

```suggestion
title: "How to Migrate your JavaScript from Sprockets to Webpacker"
```

lubc reviewed (commented):
@arielj Nice article! Just added some comments
